exporter.go

```diff
@@ -110,6 +113,42 @@ func createClientWithRetries(getClient func() (interface{}, error), retries uint
 	return nil, err
 }
 
+func parseUnixSocketAddress(listenAddress string) (string, string, error) {
+	listenAddressParts := strings.Split(listenAddress, ":")
+	listenAddressPartsLength := len(listenAddressParts)
+
+	if listenAddressPartsLength > 3 || listenAddressPartsLength < 1 {
+		return "", "", fmt.Errorf("address for unix domain socket has wrong format")
+	}
+
+	unixSocketPath := listenAddressParts[1]
+	requestPath := ""
+	if listenAddressPartsLength == 2 {
+		requestPath = listenAddressParts[2]
+	}
+	return unixSocketPath, requestPath, nil
+}
+
+func startListeningOn(listenAddress string) error {
+	if !strings.HasPrefix(listenAddress, "unix:") {
```

Comment:
Additionally, maybe we could refactor this method and make it more consistent, we could simplify the logic.

**Note**: Not tested, just to give you the main idea.

```golang
if !strings.HasPrefix(listenAddress, "unix:") {
	path, _, err := parseUnixSocketAddress(listenAddress)
	if err != nil {
		log.Fatalf("parsing unix domain socket listen address %s failed: %+v", listenAddress, err)
	}
	listener, err := net.ListenUnix("unix", &net.UnixAddr{Name: path, Net: "unix"})
} else {
	listener, err := net.Listen("tcp", listenAddress)
}
return http.Serve(listener, nil)
```

This way the method will choose the right listener depending on the address and then it will use the same code for the `http.Serve`.

Let me know what do you think.

Reply:
I think your code is cleaner. I will include it in the PR.